x-pack/plugins/task_manager/server/monitoring/workload_statistics.test.ts

```diff
@@ -477,6 +477,41 @@ describe('Workload Statistics Aggregator', () => {
       }, reject);
     });
   });
+
+  test('recovery after errors occurrs at the next interval', async () => {
+    const refreshInterval = 1000;
+
+    const taskStore = taskStoreMock.create({});
+    const logger = loggingSystemMock.create().get();
+    const workloadAggregator = createWorkloadAggregator(
+      taskStore,
+      of(true),
+      refreshInterval,
+      3000,
+      logger
+    );
+
+    return new Promise((resolve, reject) => {
+      let errorWasThrowAt = 0;
+      taskStore.aggregate.mockImplementation(async () => {
+        if (errorWasThrowAt === 0) {
+          errorWasThrowAt = Date.now();
+          throw new Error(`Elasticsearch has gone poof`);
+        } else if (Date.now() - errorWasThrowAt < refreshInterval) {
+          reject(new Error(`Elasticsearch is still poof`));
+        }
+
+        return setTaskTypeCount(mockAggregatedResult(), 'alerting_telemetry', {
+          idle: 2,
+        });
+      });
+
+      workloadAggregator.pipe(take(2), bufferCount(2)).subscribe((results) => {
```

Comment:
is the `bufferCount(2)` needed here?  I woulda thought `take(2)` would be good enough to get the first two results, no buffering needed.

Reply:
`take(2)` emits 2 events.
`bufferCount(2)` merges them into an array and calls the handler _once_ with that array instead of twice (one for each item).